app/controllers/people/person/positions/minister/edit.js

```diff
@@ -0,0 +1,85 @@
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { dropTask } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+import { combineFullAddress } from 'frontend-contact-hub/models/address';
+
+const FINANCING_CODE = {
+  SELF_FINANCED: '997073905f839ac6bafe92b76050ab0b',
+  FOD_FINANCED: '9d6f49b3d923b437ec3a91e8b5fa6885',
+};
+
+export default class PeoplePersonPositionsMinisterEditController extends Controller {
+  @service router;
+  @tracked willReceiveFinancing;
+  @tracked isCurrentPosition;
+  @tracked redirectUrl;
+
+  queryParams = ['redirectUrl'];
+
+  setup() {
+    this.willReceiveFinancing =
+      this.model.minister.financing.get('id') === FINANCING_CODE.FOD_FINANCED;
+    this.isCurrentPosition = !this.model.minister.agentEndDate;
+  }
+
+  @action
+  async clearOnCheck() {
+    this.model.minister.agentEndDate = undefined;
+    this.isCurrentPosition = true;
```

Comment:
The current setup is a bit weird. The checkbox component uses 2 way binding to update the checked state by default, which is why `this.isCurrentPosition` is set to the correct value even if it's being hardcoded to `true` here.

It's confusing when reading the code though, because it looks like it isn't supposed to work.

I would remove the 2 way binding from the component and update the state in this method instead.